worker/src/worker.rs

```diff
@@ -457,16 +442,25 @@ impl WorkerToWorker for WorkerReceiverHandler {
                 .send(batch)
                 .await
                 .map_err(|_| DagError::ShuttingDown),
-
-            WorkerMessage::BatchRequest(missing, requestor) => self
-                .tx_worker_helper
-                .send((missing, requestor))
-                .await
-                .map_err(|_| DagError::ShuttingDown),
         }
         .map(|_| anemo::Response::new(()))
         .map_err(|e| anemo::rpc::Status::internal(e.to_string()))
     }
+    async fn request_batches(
+        &self,
+        request: anemo::Request<types::WorkerBatchRequest>,
+    ) -> Result<anemo::Response<types::WorkerBatchResponse>, anemo::rpc::Status> {
+        let message = request.into_body();
+        let batches: Vec<Batch> = self
+            .store
+            .read_all(message.digests)
```

Comment:
I am thinking that maybe we should keep a TODO comment as before (`// TODO [issue #7]: Do some accounting to prevent bad actors from monopolizing our resources`) , but updated, as a byzantine node could send a huge list here and stress our node. Something to not forget about.